src/web/app/pages-student/student-profile-page/student-profile-page.component.ts

```diff
@@ -112,25 +115,41 @@ export class StudentProfilePageComponent implements OnInit {
    * Opens a modal box to upload/edit profile picture.
    */
   onUploadEdit(): void {
-    const modalRef: NgbModalRef = this.ngbModal.open(UploadEditProfilePictureModalComponent);
-    modalRef.componentInstance.profilePicLink = this.profilePicLink;
-    modalRef.result.then((formData: FormData) => {
-      if (!formData) {
-        this.statusMessageService.showWarningMessage('No photo uploaded');
-        return;
-      }
-
-      this.httpRequestService.post('/student/profilePic', {}, formData)
-          .subscribe(() => {
-            this.statusMessageService.showSuccessMessage('Your profile picture has been saved successfully');
-
-            // force reload
-            const timestamp: number = (new Date()).getTime();
-            this.profilePicLink = `${this.backendUrl}/webapi/student/profilePic?${timestamp}&user=${this.id}`;
-          }, (response: ErrorMessageOutput) => {
-            this.statusMessageService.showErrorMessage(response.error.message);
-          });
-    }, () => {});
+    this.getProfilePicture()
+        // Return null if no picture is found
+        .pipe(
+            catchError(() => of(null)),
+        )
+        // Open Modal and wait for user to upload picture
+        // Throw an error if there is no picture is uploaded
+        .pipe(
+            flatMap((image: Blob | null) => {
+              const modalRef: NgbModalRef = this.ngbModal.open(UploadEditProfilePictureModalComponent);
+              modalRef.componentInstance.image = image;
+
+              return from(modalRef.result);
+            }),
+            catchError(() => throwError({
+              error: {
```

Comment:
The error message format is for the convenient of which handler?

Reply:
It is for the `response` object of type `ErrorMessageOutput` in line 150

Reply:
It should not be an error as users just dismiss the model?

Reply:
That makes a lot more sense, will do!

Reply:
I can't find a clean way to terminate a pipe in RxJs, so in the new implementation I have passed `null` down the pipe to indicate that the user didn't upload any picture.

Would it be better to throw an error when the user does not upload any picture, but not display the error message to the user? Or is there a better alternative.

Reply:
> Would it be better to throw an error when the user does not upload any picture, but not display the error message to the user?

I think so.